tgui/packages/tgui/interfaces/NtosCard.js

```diff
@@ -17,44 +17,202 @@ export const NtosCard = (props, context) => {
+                extraButtons={
+                  <Button.Confirm
+                    content="Terminate Employment"
+                    confirmContent="Fire Employee?"
+                    color="bad"
+                    onClick={() => act('PRG_terminate')} />
+                }
+                accessMod={(ref, wildcard) => act('PRG_access', {
+                  access_target: ref,
+                  access_wildcard: wildcard,
+                })} />
+            </Box>
+          )}
+        </Stack.Item>
+      </Stack>
+    </>
+  );
+};
+
+const IDCardTabs = (props, context) => {
+  const [
+    selectedTab,
+    setSelectedTab,
+  ] = useSharedState(context, "selectedTab", "login");
+
+  return (
+    <Tabs vertical fill>
+      <Tabs.Tab
+        minWidth={"100%"}
+        altSelection
+        selected={"login" === selectedTab}
+        color={("login" === selectedTab) ? "green" : "default"}
+        onClick={() => setSelectedTab("login")}>
+        Login ID
+      </Tabs.Tab>
+      <Tabs.Tab
+        minWidth={"100%"}
+        altSelection
+        selected={"modify" === selectedTab}
+        color={("modify" === selectedTab) ? "green" : "default"}
+        onClick={() => setSelectedTab("modify")}>
+        Target ID
+      </Tabs.Tab>
+    </Tabs>
+  );
+};
+
+export const IDCardLogin = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    authenticatedUser,
+    has_id,
+    have_printer,
+    authIDName,
+  } = data;
+
+  return (
+    <Section
+      title="Login"
+      buttons={(
+        <>
+          <Button
+            icon="print"
+            content="Print"
+            disabled={!have_printer || !has_id}
+            onClick={() => act('PRG_print')} />
+          <Button
+            icon={authenticatedUser ? "sign-out-alt" : "sign-in-alt"}
+            content={authenticatedUser ? "Log Out" : "Log In"}
+            color={authenticatedUser ? "bad" : "good"}
+            onClick={() => {
+              act(authenticatedUser ? 'PRG_logout' : 'PRG_authenticate');
+            }} />
+        </>
+      )}>
+      <Stack wrap="wrap">
+        <Stack.Item width="100%">
+          <Button
+            fluid
+            ellipsis
+            icon="eject"
+            content={authIDName}
+            onClick={() => act('PRG_ejectauthid')} />
+        </Stack.Item>
+        <Stack.Item width="100%" mt={1} ml={0}>
+          Login: {authenticatedUser || "-----"}
+        </Stack.Item>
+      </Stack>
+    </Section>
+  );
+};
+
+const IDCardTarget = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    authenticatedUser,
+    id_rank,
+    id_owner,
+    has_id,
+    id_name,
+    id_age,
+  } = data;
+
+  return (
+    <Section title="Modify ID">
+      <Button
+        width="100%"
+        ellipsis
+        icon="eject"
+        content={id_name}
+        onClick={() => act('PRG_ejectmodid')} />
+      {(has_id && authenticatedUser) && (
```

Comment:
Add !! in front of the first condition to cast it to boolean, then you won't need || false at the very end.

Reply:
Done.